src/GitVersionCore/Extensions/GitExtensions.cs

```diff
@@ -43,24 +43,24 @@ public static bool IsSameBranch(this Branch branch, Branch otherBranch)
         /// <summary>
         /// Exclude the given branches (by value equality according to friendly name).
         /// </summary>
-        public static IEnumerable<BranchCommit> ExcludingBranches(this IEnumerable<BranchCommit> branches, IEnumerable<Branch> branchesToExclude)
+        public static IEnumerable<BranchCommit> ExcludingBranches(this IEnumerable<BranchCommit> branches, IEnumerable<IBranch> branchesToExclude)
```

Comment:
Any reason not to provide `ExcludingBranches` as a method on the `IBranchCollection` interface?

Reply:
Agree, I will change that in a subsequent PR